events/companies/views.py

```diff
@@ -0,0 +1,234 @@
+NO_CONTENT = HttpResponse(status=204)
+CREATED = JsonResponse({'success': True}, status=201)
+
+
+class CompanyView(View):
+
+    def get(self, request, company_id=None):
+        if not company_id:
+            if not request.user.is_superuser:
+                return PERMISSION_DENIED
+            companies = Company.get_all()
+            response = {"companies": [model_to_dict(company) for company in companies]}
+            return JsonResponse(response, status=200)
+
+        company = Company.get_by_id(company_id)
+        if not company:
+            return COMPANY_NOT_EXISTS
+        if Company.get_user_company(request) != company:
+            if not request.user.is_superuser:
+                return PERMISSION_DENIED
+        response = model_to_dict(company)
+        response['teams'] = Company.get_teams(company_id)
+        return JsonResponse(response, status=200)
+
+    def post(self, request):
+        if not request.user.is_superuser:
+            return PERMISSION_DENIED
+        new_company_data = json.loads(request.body.decode())
+        company_form = CompanyForm(new_company_data)
+        errors = company_form.errors
+        user = User.get_by_id(new_company_data.get('admin'))
+        errors = CompanyView.validate_admin_field(user, errors, new_company_data)
+        if not company_form.is_valid() or errors:
+            return JsonResponse({'success': False, 'errors': errors}, status=400)
+        new_company_data['admin'] = user
+        Company.objects.create(**new_company_data)
+        return CREATED
+
+    def put(self, request, company_id):
+        company = Company.get_by_id(company_id)
+        if not company:
+            return COMPANY_NOT_EXISTS
+        if not CompanyView.check_company_rights(request, company):
+                return PERMISSION_DENIED
+        upd_company_data = json.loads(request.body.decode())
+        company_form = CompanyForm(upd_company_data)
+        errors = company_form.errors
+        upd_admin = User.get_by_id(upd_company_data.get('admin'))
+        if upd_admin and upd_admin != company.admin:
+            if not request.user.is_superuser:
+                return PERMISSION_DENIED
+            errors = CompanyView.validate_admin_field(upd_admin, errors, upd_company_data, company)
+        else:
+            upd_admin = company.admin
+        if not company_form.is_valid() or errors:
+            return JsonResponse({'success': False, 'errors': company_form.errors}, status=400)
+        upd_company_data['admin'] = upd_admin
+        Company.objects.filter(pk=company_id).update(**upd_company_data)
+        return CREATED
+
+    def delete(self, request, company_id):
+        company = Company.get_by_id(company_id)
+        if not CompanyView.check_company_rights(request, company):
+                return PERMISSION_DENIED
+        company.delete()
+        return NO_CONTENT
+
+    @staticmethod
+    def check_company_rights(request, company):
```

Comment:
@VKotsovskyi By now `check_company_rights` method is used in all of the three views.